pandas/_libs/tslibs/fields.pyx

```diff
@@ -113,6 +117,40 @@ def get_date_name_field(ndarray[int64_t] dtindex, object field):
             dt64_to_dtstruct(dtindex[i], &dts)
             dow = dayofweek(dts.year, dts.month, dts.day)
             out[i] = _dayname[dow]
+    if field == 'day_name':
+        if time_locale is None:
+            _dayname = np.array(['monday', 'tuesday', 'wednesday', 'thursday',
+                                 'friday', 'saturday', 'sunday'],
+                                dtype=np.object_)
+        else:
+            with set_locale(time_locale, locale.LC_TIME):
```

Comment:
this function (the locale conversion of the names) should be in ccalandar.pyx
just have a function which takes the locale (default None), so this is much simpler here